File: ts/problems/66-plus-one.test.ts
import { describe, expect, test } from "vitest";
import { plusOne } from "./66-plus-one";

describe("plusOne", () => {
  test.each([
    {
      digits: [0],
      want: [1],
    },
    {
      digits: [1, 2, 3],
      want: [1, 2, 4],
    },
    {
      digits: [4, 3, 2, 1],
      want: [4, 3, 2, 2],
    },
    {
      digits: [9],
      want: [1, 0],
    },
  ])("given $digits we want $want", ({ digits, want }) => {
    expect(plusOne(digits)).toStrictEqual(want);
  });
});

File: go/problems/66_plus_one_test.go
package main

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestPlusOne(t *testing.T) {
	tests := []struct {
		digits []int
		want   []int
	}{
		{
			digits: []int{0},
			want:   []int{1},
		},
		{
			digits: []int{1, 2, 3},
			want:   []int{1, 2, 4},
		},
		{
			digits: []int{4, 3, 2, 1},
			want:   []int{4, 3, 2, 2},
		},
		{
			digits: []int{9},
			want:   []int{1, 0},
		},
	}

	for _, tt := range tests {
		t.Run(fmt.Sprintf("given %v we want %v", tt.digits, tt.want), func(t *testing.T) {
			got := plusOne(tt.digits)
			assert.Equal(t, tt.want, got, "got %v want %v", got, tt.want)
		})
	}
}
